s3parq/publish_parq.py

```diff
@@ -175,13 +180,13 @@ def _gen_parquet_to_s3(bucket: str, key: str, dataframe: pd.DataFrame,
 
     try:
         table = pa.Table.from_pandas(
-            df=dataframe, schema=_parquet_schema(dataframe), preserve_index=False)
+            df=dataframe, schema=_parquet_schema(dataframe, custom_redshift_columns=custom_redshift_columns), preserve_index=False)
     except pa.lib.ArrowTypeError:
         logger.warn(
             "Dataframe conversion to pyarrow table failed, checking object columns for mixed types")
         dataframe_dtypes = dataframe.dtypes.to_dict()
         object_columns = [
-            col for col, col_type in dataframe_dtypes.items() if col_type == "object"]
+            col for col, col_type in dataframe_dtypes.items() if col_type == "object" and "Decimal(" not in str(dataframe[col].values)[:10]]
```

Comment:
this line is oddly specific.  maybe a quick comment about why we check for "Decimal(".  also, "Decimal(" is only 8 chars long but you get a 10 char slice.

Reply:
hey! This is here because we want to skip conversion to strings if the column contains decimal objects. Without this, decimal objects will be improperly converted. I'll drop a comment and fix it up, that's a good call as it isn't handled eloquently.